hashintel/hash
# Entity Type Editor: Persist "Required" and "Allow Multiple Values"

## 🌟 What is the purpose of this PR?

This PR persists the value of a property's "required" and "allow multiple values" fields in the entity type editor.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/0/1203265599241346/f) _(internal)_
- [[3] Allow multiple values min/max UI (type-editor)](https://app.asana.com/0/0/1203198755412034/f) (_internal_)

## 🔍 What does this change?

- Added a Popper to set the Minimum and Maximum number of items;
- Added a Pill to reflect the minimum and maximum number of items of a property;
- Added the required and array fields to the entity type save/update requests.

## 🐾 Next steps

Implement persisting "Default Value" -> Waiting on this [task](https://app.asana.com/0/1200211978612931/1203242624811525/f) _(internal)_

## ❓ How to test this?

1. Create a new entity type;
2. Add a property and change its "required" and "allow multiple values" (as well as the Minimum and Maximum number of items);
3. Save the entity type;
4. Reload the page to check if the changes persist.

## 📹 Demo

![persist](https://user-images.githubusercontent.com/37453800/198894360-2fca6a99-9fdc-43b2-b670-a0c11fd723e4.gif)


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203265599241346
